### ISSUE TITLE
IN operator not working for SQLite

### ISSUE THREAD

**Issue by rotmoset** [fsprojects/SQLProvider#167]
When using the IN operator ( |=| ), the generated query looks like so:

```SQL
select stuff from tables where column in (@param1)
```

this is not correct (for sqlite at least) as every list element needs to be inserted as a separate parameter.

More info here: 
http://stackoverflow.com/questions/2377506/pass-array-parameter-in-sqlcommand
http://stackoverflow.com/questions/32485431/c-sharp-sqlite-how-to-pass-bound-parameters-via-an-array-or-list

I did a quickfix, in Providers.SQLite.fs like so:
```fsharp
~~(sprintf "%s%s" prefix <|
	let listOperation op =
		let paras =
			match paras.[0].Value with
			| :? array<obj> as arr -> arr 
			| :? seq<obj> as seq -> seq |> Array.ofSeq
			| _ -> failwith "Unsupported list type"
			|> Array.map createParam  
		let text = String.Join(",",paras |> Array.map (fun p -> p.ParameterName))
		Array.iter parameters.Add paras
		(sprintf "[%s].[%s] %s (%s)") alias col op text
	match operator with
	| FSharp.Data.Sql.IsNull -> (sprintf "[%s].[%s] IS NULL") alias col 
	| FSharp.Data.Sql.NotNull -> (sprintf "[%s].[%s] IS NOT NULL") alias col 
	| FSharp.Data.Sql.In -> listOperation "IN"
	| FSharp.Data.Sql.NotIn -> listOperation "NOT IN"
	| _ -> 
		parameters.Add paras.[0]
		(sprintf "[%s].[%s]%s %s") alias col 
			(operator.ToString()) paras.[0].ParameterName)
```

I can submit a PR, but I don't know if my fix is "in the right place" and if it's safe as I don't know enough of the source code in this project.

**Comment by rotmoset** [comment]
Please disregard the code in the above comment, I have investigated the issue further and have created a PR that fixes the root issue.

https://github.com/fsprojects/SQLProvider/pull/168

Text copied from PR:

To detect arrays being passed to the IN and NOT IN operators, a runtime type check was performed against "string array", however what was probably changed in this [changeset](https://github.com/fsprojects/SQLProvider/commit/35df630919a662fe368b0768e1d93d78b4059a40) the array actually being passed is of type "obj array". This PR fixes this check for all supported databases (this has actually been fixed multiple times before for individual vendors, see [this](https://github.com/fsprojects/SQLProvider/commit/0f0e3a8a8a26567927b636d634e66e62d3e28ecc) and [this](https://github.com/fsprojects/SQLProvider/commit/f7d379bc2010dd431e7eff69b33406d8f7e9732b), but this PR fixes this in uniform way).

**Comment by colinbull** [comment]
Fixed in #168
